src/WebJobs.Script.WebHost/Diagnostics/ConsoleWriter.cs

```diff
@@ -0,0 +1,177 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Text;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics
+{
+    internal class ConsoleWriter
+    {
+        private const int DefaultBufferSize = 1000;
```

Comment:
How many function calls is this? e.g. do we have 1 line per call, 5, 10? I'd say you'd want to aim for something like 1-10,000 * per-request in the count here (with comment indicating it). 1,000 just isn't a lot under load and I'm not sure this would behave near as well as it could in multi-core scenarios going so low - thoughts?

Reply:
Good call! We generate 8 lines for a typical out of proc execution. I tested buffer sizes of 1K, 8K and 80K lines. The P99 latency definitely improves jumping from 1K to 8K but then it plateaus. There is some sign of RPS improvement when going from 8K to 80K but I am worried about the memory usage impact, especially in the context of containers that are allocated smaller amounts of memory. We could eventually optimize further by specifying different buffer sizes for different container sizes but I don't want to get into that right now. I'll send an update bumping this to 8K.

All results below are the median of 3 runs.

1K buffer, batched:
```
Bombarding http://py39-noai-mesh-perfcollect-consolebuffer-buffer1000-batched.app:80/api/helloworld for 1m0s using 30 connection(s)
Statistics        Avg      Stdev        Max
  Reqs/sec       393.59     254.27    1631.41
  Latency       76.23ms    32.45ms   784.71ms
  Latency Distribution
     50%    70.00ms
     75%    79.00ms
     90%    89.00ms
     95%    97.00ms
     99%   336.62ms
  HTTP codes:
    1xx - 0, 2xx - 23625, 3xx - 0, 4xx - 0, 5xx - 0
    others - 0
  Throughput:   121.39KB/s
```
8K buffer, batched:
```
Bombarding http://py39-noai-mesh-perfcollect-consolebuffer-buffer8000-batched.app:80/api/helloworld for 1m0s using 30 connection(s)
Statistics        Avg      Stdev        Max
  Reqs/sec       413.62     228.98    1578.89
  Latency       72.51ms     7.91ms   309.00ms
  Latency Distribution
     50%    72.00ms
     75%    80.73ms
     90%    89.00ms
     95%    94.94ms
     99%   110.59ms
  HTTP codes:
    1xx - 0, 2xx - 24837, 3xx - 0, 4xx - 0, 5xx - 0
    others - 0
  Throughput:   127.61KB/s
```

80K buffer, batched:
```
Bombarding http://py39-noai-mesh-perfcollect-consolebuffer-buffer80000-batched.app:80/api/helloworld for 1m0s using 30 connection(s)
Statistics        Avg      Stdev        Max
  Reqs/sec       427.36     250.94    1799.71
  Latency       70.21ms     8.27ms   186.00ms
  Latency Distribution
     50%    69.99ms
     75%    78.35ms
     90%    87.99ms
     95%    94.69ms
     99%   111.02ms
  HTTP codes:
    1xx - 0, 2xx - 25640, 3xx - 0, 4xx - 0, 5xx - 0
    others - 0
  Throughput:   132.23KB/s
```

This result makes sense to me, given that the current implementation will block for up to 1 second waiting on the buffer to become available before falling back to the console. In the 1K case, the P99 latency really looks like its impacted by waits on the buffer to be cleared.